### PR TITLE
Metagenomics

### PR DESCRIPTION
**Pull request by josenavas**
This adds the code needed to perform all computations for a metagenomics run. The original code is from @tanaes ipython notebook located [here](https://github.com/tanaes/metagenomics_pooling_notebook/blob/master/metagenomics_pipeline_MiniPiconorm.ipynb). I've adapted the DB to store the extra information needed and the functions interface to better match the data structures present here.

@tanaes, if you can take a quick look that would be great!

### CODE REVIEW

**Comment by tanaes** [comment]
Awesome! I'll take a look at this. 

-j

**Comment by jdereus** [comment]
can we get a fresh deploy on maptest please?

**Comment by josenavas** [comment]
@jdereus this did not change anything on the interface - this was all backend for the shotgun side - I don't see why would be useful to have a fresh deploy.

**Comment by jdereus** [comment]
Because the current deploy still has issues that are not resolved that limit core functionality for testing.  You are expecting to do a code sprint, you should have an updated deploy out there.  The db has changed, the code has changed.  How many times have we seen unintended consequences from code and db changes?

File: labman/db/process.py [hunk 1]
```diff
@@ -615,6 +803,127 @@ def water_lot(self):
...
+                # For the source well we need to take a look at the gdna comp
+                gdna_comp = composition.gdna_composition
+                wells.append(gdna_comp.container.well_id)
+                dest_wells.append(well.well_id)
+                # For the sample name we need to check the sample composition
+                sample_comp = gdna_comp.sample_composition
+                sample_names.append(sample_comp.content)
+                # For the DNA concentrations we need to look at
+                # the quantification process
+                dna_concs.append(concentrations[gdna_comp])
```
Comment (tanaes) [inline labman/db/process.py]: where is it that `reformat` can be specified in this process?

File: labman/db/process.py [hunk 2]
```diff
@@ -633,6 +942,101 @@ class LibraryPrepShotgunProcess(Process):
...
+                                    ON np.process_id =
+                                        c.latest_upstream_process_id
+                                JOIN qiita.well USING (container_id)
+                                WHERE plate_id = %s))
+                     RETURNING library_prep_shotgun_process_id"""
+            TRN.add(sql, [process_id, kappa_hyper_plus_kit.id, stub_lot.id,
+                          plate.id])
+            instance = cls(TRN.execute_fetchlast())
+
+            # Get the primer set for the plates
```
Comment (tanaes) [inline labman/db/process.py]: I don't quite understand what is going on here ... why just i5 being used?

File: labman/db/process.py [hunk 3]
```diff
@@ -665,6 +1069,115 @@ def normalization_process(self):
...
+            lib_comp = well.composition
+            sample_comp = lib_comp.normalized_gdna_composition\
+                .gdna_composition.sample_composition
+            sample_names.append(sample_comp.content)
+            # Retrieve all the information about the indices
+            i5_comp = lib_comp.i5_composition.primer_set_composition
+            i5_well = i5_comp.container
+            indices['i5 name'][idx] = i5_comp.external_id
+            indices['i5 plate'][idx] = i5_well.plate.external_id
+            indices['i5 sequence'][idx] = i5_comp.barcode
```
Comment (tanaes) [inline labman/db/process.py]: maybe want to rename that property `i5_comp.sequence` to differentiate from the plate barcode?

**Pull request merged**